### DISCUSSION TITLE
Internalization support

### DISCUSSION THREAD

**Discussion by romanzy313** [solidjs/solid-start#330] [General]
Hello,

I am looking forward to start using solidjs for a web project. It really very different from traditional server/client and I love it very much. But I need internalization support for the website.

How hard would it be to integrate i18next into solidjs? Load translations on the server, do server-side rendering and api. Then dynamically load translation on the client as needed. I am interested in the following routing strategies:

1) subdomain (en.website.com/, es.website.com/) 
2) suburl (website.com/en/, website.com/es/)

Any starting points or already made solutions for the solid-start ecosystem?

**Comment by izznat** [comment]
For sub-url we need to support optional route param first. Btw, optional route param also useful for something like page embed.
